src/full/Agda/Compiler/MAlonzo/Compiler.hs

```diff
@@ -226,20 +226,24 @@ imports = (hsImps ++) <$> imps where
   uniq = List.map head . List.group . List.sort
 
 -- Should we import MAlonzo.RTE.Float
-data UsesFloat = YesFloat | NoFloat
-  deriving (Eq, Show)
+newtype UsesFloat = UsesFloat Bool deriving (Eq, Show)
+
+pattern YesFloat :: UsesFloat
+pattern YesFloat = UsesFloat True
+
+pattern NoFloat :: UsesFloat
+pattern NoFloat  = UsesFloat False
 
 instance Semigroup UsesFloat where
-  NoFloat  <> i        = i
-  i        <> NoFloat  = i
-  YesFloat <> YesFloat = YesFloat
+  UsesFloat a <> UsesFloat b = UsesFloat (a || b)
 
 instance Monoid UsesFloat where
   mempty  = NoFloat
   mappend = (<>)
 
 mazRTEFloatImport :: UsesFloat -> [HS.ImportDecl]
-mazRTEFloatImport i = [ HS.ImportDecl mazRTEFloat True Nothing | i == YesFloat ]
+mazRTEFloatImport (UsesFloat True)  = [ HS.ImportDecl mazRTEFloat True Nothing ]
```

Comment:
Seems like you could exploint the `Bool` here:
```haskell
mazRTEFloatImport (UsesFloat b)  = [ HS.ImportDecl mazRTEFloat True Nothing | b ]
```

Reply:
Yes, you're right. It felt to me that the use of a list comprehension to generate a one-element list (as it was originally) was less clear to read than case-splitting. A few more mental hoops to jump through to see that it was a 0/1 element list. But as a one-liner it doesn't matter much, I'm happy to change it.

Reply:
(Done!)